internal/multicluster/internal/types/validation_exported_services_ce.go

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+//go:build !consulent
+
+package types
+
+import (
+	"fmt"
+	"github.com/hashicorp/consul/internal/resource"
+	multiclusterv1alpha1 "github.com/hashicorp/consul/proto-public/pbmulticluster/v1alpha1"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+	"github.com/hashicorp/go-multierror"
+)
+
+func ValidateExportedServicesEnterprise(res *pbresource.Resource, exportedService *multiclusterv1alpha1.ExportedServices) error {
+	var hasSetEnterpriseFeatures bool
+
+	invalidFields := make([]string, 0)
+
+	if res.Id != nil && res.Id.Tenancy != nil && (res.Id.Tenancy.Namespace != "default" || res.Id.Tenancy.Partition != "default") {
+		if res.Id.Tenancy.Namespace != "default" {
+			invalidFields = append(invalidFields, "namespace")
+		} else {
+			invalidFields = append(invalidFields, "partition")
+		}
+		hasSetEnterpriseFeatures = true
+	}
```

Comment:
We don't need to worry about validating tenancy. It is done in resource service here:
- https://github.com/hashicorp/consul/blob/main/agent/grpc-external/services/resource/write.go#L45-L50
- https://github.com/hashicorp/consul/blob/main/agent/grpc-external/services/resource/server_ce.go#L14